api/database/seeders/SkillSeeder.php

```diff
@@ -2,28 +2,1925 @@
+                'skill_name_fr' => 'Respect de la diversité',
+                'skill_definition_en' => 'Appreciate the unique qualities and views people from different backgrounds bring to a team.',
+                'skill_definition_fr' => 'Apprécier (encourager? célébrer?) les qualités et les points de vue uniques que des personnes d’origines différentes apportent à une équipe.',
+                'keywords_en' => 'Inclusive; Accepting of differences; Cultural Competence',
+                'keywords_fr' => 'Inclusivité;  Acceptation des différences;  Compétence culturelle'
+            ],
+            [
+
+                'skill_family_en' => 'Working in Government',
+                'skill_name_en' => 'Working on a Distributed Team',
+                'skill_name_fr' => 'Travailler dans une équipe distribuée',
+                'skill_definition_en' => 'Use available resources to communicate, collaborate, and work in a way that engages team members in distributed locations. ',
+                'skill_definition_fr' => 'Utiliser les ressources disponibles pour communiquer, collaborer et travailler de manière à impliquer les membres de l’équipe dans des lieux distribués.',
+                'keywords_en' => 'Remote Work; Work From Home',
+                'keywords_fr' => 'Travail à distance;  Travail à domicile'
+            ],
+        ];
+
+        // as close as we have to a unique identifier in the data
+        $englishNames = array_unique(
+            array_map(function ($record) {
+                return trim($record["skill_name_en"]);
+            },
+            $providedData
+        ));
+
+        $reshapedData = array_map(
+            function ($englishName) use ($providedData) {
+                // could be more than one if skill is part of multiple skill families
+                $matchingRecords = array_filter(
+                    $providedData,
+                    function($record) use ($englishName) {
+                        return trim($record['skill_name_en']) == $englishName;
+                    }
+                );
+
+                // they *should* all be the same except for skill family so we'll build the model on the first one
+                $firstMatchingRecord = reset($matchingRecords);
+
+                // Take the provided data and reshape it to our data model
+                $model = [
+                    'key' => KeyStringHelpers::toKeyString($englishName), // no key provided so making our own slug
+                    'name' => [
+                        'en' => trim($firstMatchingRecord['skill_name_en']),
+                        'fr' => trim($firstMatchingRecord['skill_name_fr'])
+                    ],
+                    'description' => [ // no descriptions provided so reusing name
+                        'en' => trim($firstMatchingRecord['skill_definition_en']),
+                        'fr' => trim($firstMatchingRecord['skill_definition_fr'])
```

Comment:
this isn't recycling name though?
in some cases there isn't a skill_definition provided, should there be a placeholder message placed somewhere? like on the frontend component `useGetQuery -> if (!data.description.en) {then placeholder}` just to avoid having blanks

Reply:
Yep, it's definitely recycling the name.  The problem with a placeholder is that there's nothing to hold the place for.  :laughing: This is all the data we were provided.  Since the description field is non-nullable I figured that displaying the name a second time would be less confusing than displaying a single space or some other arbitrary constant.